### ISSUE TITLE
Setting --experimental_sandbox_limits=cpu=N fails with NoSuchElementException

### ISSUE THREAD

**Issue by arjantop-cai** [bazelbuild/bazel#24680]
### Description of the bug:

When running build with the flag `experimental_sandbox_limits` bazel fails with `NoSuchElementException`.
Looks like bazel is trying to read `cpu.max` from the cgroup and uses a skip regex `.*\\s` which causes the scanner to skip all input as `.*` matching is greedy and the value ends with a newline.

Example hexdump of some `cpu.max`:
```
0000000 155 141 170 040 061 060 060 060 060 060 012                    
0000000   m   a   x       1   0   0   0   0   0  \n                    
000000b
```

Command:
```
bazel build --incompatible_use_new_cgroup_implementation --experimental_sandbox_limits=cpu=1 //some:target
```
Error:
```
java.lang.RuntimeException: Unrecoverable error while evaluating node 'ActionLookupData0{actionLookupKey=ConfiguredTargetKey{label=@@rules_go+//:stdlib, config=BuildConfigurationKey[5e8dec90704ee94878bb1429b2c5a6516db1bd61488e8fd29aebbe88b5f273b8]}, actionIndex=0}' (requested by nodes 'File:[[<execution_root>]bazel-out/aarch64-fastbuild-ST-bf922961484a/bin]external/rules_go+/stdlib_/pkg')
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:547)
	at com.google.devtools.build.lib.concurrent.AbstractQueueVisitor$WrappedRunnable.run(AbstractQueueVisitor.java:435)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
Caused by: java.util.NoSuchElementException
	at java.base/java.util.Scanner.throwFor(Unknown Source)
	at java.base/java.util.Scanner.next(Unknown Source)
	at java.base/java.util.Scanner.nextInt(Unknown Source)
	at java.base/java.util.Scanner.nextInt(Unknown Source)
	at com.google.devtools.build.lib.sandbox.cgroups.controller.v2.UnifiedCpu.setCpus(UnifiedCpu.java:45)
	at com.google.devtools.build.lib.sandbox.cgroups.VirtualCgroupFactory.create(VirtualCgroupFactory.java:73)
	at com.google.devtools.build.lib.sandbox.LinuxSandboxedSpawnRunner.prepareSpawn(LinuxSandboxedSpawnRunner.java:334)
	at com.google.devtools.build.lib.sandbox.AbstractSandboxSpawnRunner.exec(AbstractSandboxSpawnRunner.java:112)
	at com.google.devtools.build.lib.sandbox.SandboxModule$SandboxFallbackSpawnRunner.exec(SandboxModule.java:479)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:158)
	at com.google.devtools.build.lib.exec.AbstractSpawnStrategy.exec(AbstractSpawnStrategy.java:118)
	at com.google.devtools.build.lib.exec.SpawnStrategyResolver.exec(SpawnStrategyResolver.java:45)
	at com.google.devtools.build.lib.analysis.actions.SpawnAction.execute(SpawnAction.java:253)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.executeAction(SkyframeActionExecutor.java:1170)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor$ActionRunner.run(SkyframeActionExecutor.java:1075)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.runStateMachine(ActionExecutionState.java:166)
	at com.google.devtools.build.lib.skyframe.ActionExecutionState.getResultOrDependOnFuture(ActionExecutionState.java:95)
	at com.google.devtools.build.lib.skyframe.SkyframeActionExecutor.executeAction(SkyframeActionExecutor.java:559)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.checkCacheAndExecuteIfNeeded(ActionExecutionFunction.java:928)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.computeInternal(ActionExecutionFunction.java:375)
	at com.google.devtools.build.lib.skyframe.ActionExecutionFunction.compute(ActionExecutionFunction.java:216)
	at com.google.devtools.build.skyframe.AbstractParallelEvaluator$Evaluate.run(AbstractParallelEvaluator.java:467)
	... 7 more
```

### Which category does this issue belong to?

Core, Local Execution

### What's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Run:
```
bazel build --incompatible_use_new_cgroup_implementation --experimental_sandbox_limits=cpu=1 //some:target
```

### Which operating system are you running Bazel on?

Linux

### What is the output of `bazel info release`?

release 8.0.0

**Comment by fmeum** [comment]
@AlessandroPatti

**Comment by zhengwei143** [comment]
cc @bigelephant29

**Comment by iancha1992** [comment]
A fix for this issue has been included in [Bazel 8.0.1 RC1](https://github.com/bazelbuild/bazel/releases/tag/8.0.1rc1). Please test out the release candidate and report any issues as soon as possible.
If you're using Bazelisk, you can point to the latest RC by setting USE_BAZEL_VERSION=8.0.1rc1. Thanks!
